Warehouse.Persistence.EF/Persistence/Repositories/BasketRepository.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Warehouse.Application.Common.Interfaces.Persistence;
+using Warehouse.Domain.Entities;
+using Warehouse.Domain.Exceptions;
+using Warehouse.Persistence.Abstractions;
+using Warehouse.Persistence.EF.Persistence.Contexts;
+
+namespace Warehouse.Persistence.EF.Persistence.Repositories;
+
+public class BasketRepository : GenericRepository<Basket>, IBasketRepository
+{
+    public BasketRepository(WarehouseDbContext dbContext) : base(dbContext)
+    {
+    }
+
+    public async Task<Basket> GetSingleBasketByUserIdAsync(Guid userId)
+    {
+        var result = await _dbContext.Set<Basket>()
+            .Include(b => b.BasketLines)
+            .FirstOrDefaultAsync(o => o.UserId == userId);
```

Comment:
You can use SingleAsync and wrap in try/catch and handle invalideoprationexeption and rethrow with your custom and eliminate this` if  == null` or `if is null`